### ISSUE TITLE
Change kubectl.sh from a wrapper to an initialization script

### ISSUE THREAD

**Issue by bgrant0607** [kubernetes/kubernetes#8817]
It shouldn't be necessary to wrap the kubectl command with a wrapper script. It should be possible to just source kubectl.sh to set up the path, context, whatever, and then invoke the kubectl binary directly.

**Comment by bgrant0607** [comment]
The difference between kubectl and cluster/kubectl.sh was surfaced in the friction study. Eliminating kubectl.sh would help.

**Comment by bgrant0607** [comment]
cc @krousey

**Comment by bgrant0607** [comment]
cc @jlowdermilk

**Comment by jlowdermilk** [comment]
I agree. `cluster/kubectl.sh` should just find the appropriate binary and update your PATH. Once #9227 is merged, the only work required is the tedious chore of going through the code base and switching every instance of `cluster/kubectl.sh` to `kubectl` (along with a `source cluster/kubectl.sh` where needed).

**Comment by bgrant0607** [comment]
What's left to do here?

**Comment by bgrant0607** [comment]
And is anyone working on it?

**Comment by bgrant0607** [comment]
cc @lavalamp

**Comment by lavalamp** [comment]
It's worth noting a bad interaction with gcloud-- now that our docs just say `kubectl` and since we don't have any sort of installer, nearly everyone is using the `kubectl` that gcloud ships.

Having a `make install-client` command would be great, and having our build script finish with a `which kubectl` and a warning if it's some old one would be great.

**Comment by jlowdermilk** [comment]
I like the idea of adding a `make install-client` command. Perhaps it can find the appropriate platform binary and add a symlink to it under `/usr/local/bin`?

Other than that, the only thing left to do is go through scripts/examples and replace all occurrences of `cluster/kubectl.sh` with `kubectl`, and call whatever client install command we settle on.

**Comment by ihmccreery** [comment]
FWIW, the half-dozen or so ways we run `kubectl` from various parts of our repo is making is very difficult to work on testing version-skewed `kubectl`.  I might be forced to take this issue on.  Looking into it...

**Comment by ihmccreery** [comment]
Notes for my own sanity on how tests decide what `kubectl` to run:

- [ ] `hack/e2e.go` overrides `KUBECTL` (it doesn't respect an already-set env variable) and then also manually calls `"cluster/kubectl.sh"` later. (https://github.com/kubernetes/kubernetes/blob/master/hack/e2e.go#L95, https://github.com/kubernetes/kubernetes/blob/master/hack/e2e.go#L123)
  - [ ] `hack/e2e.go` calls some number of scripts that may or may not ignore or override `KUBECTL` (e.g. https://github.com/kubernetes/kubernetes/blob/master/hack/e2e-internal/e2e-up.sh respects it, https://github.com/kubernetes/kubernetes/blob/master/test/kubemark/common.sh#L30 overrides it)
- [ ] `hack/ginkgo-e2e.sh` adds to `PATH` to be able to call `kubectl` directly as `kubectl` as the default argument for `KubectlPath` (https://github.com/kubernetes/kubernetes/blob/master/hack/ginkgo-e2e.sh#L91, https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go#L1489)
- [ ] `test/e2e/kubectl.go` generally uses `RunKubectlOrDie` which refers to `KubectlPath`, except in "should support exec through an HTTP proxy", which instead uses `findBinary` as defined in `kubectl.go`, and seems to build `kubectl` on the fly, if I'm reading it right (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl.go#L272, https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl.go#L1517)

**Comment by bgrant0607** [comment]
We need to undo #9545

**Comment by bgrant0607** [comment]
A search shows a number of uses:
https://github.com/kubernetes/kubernetes/search?utf8=%E2%9C%93&q=%22kubectl.sh%22&type=Code

We need to make it clear to users that this is deprecated (print the warning, remove it from docs), and ensure we can eventually delete the copy of the script in /cluster, by eliminating our uses of it.

**Comment by smarterclayton** [comment]
Reassigning to another sig.

**Comment by falenn** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  curtis.l.bates@gmail.com
    Domain imwiz.com has exceeded the max emails per hour (170/150 (113%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from github-smtp2-ext8.iad.github.net ([192.30.252.199]:35615 helo=github-smtp2a-ext-cp1-prd.iad.github.net)
	by box969.bluehost.com with esmtps (TLSv1.2:ECDHE-RSA-AES256-GCM-SHA384:256)
	(Exim 4.87)
	(envelope-from <noreply@github.com>)
	id 1dGU89-000QPi-0l
	for dev@imwiz.com; Thu, 01 Jun 2017 11:43:29 -0600
Date: Thu, 01 Jun 2017 10:36:13 -0700
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=github.com;
	s=pf2014; t=1496338573;
	bh=WT3s4e0QRtfXTLOQ69MCInJB5gUbxuASXwvz1w7DeQY=;
	h=From:Reply-To:To:Cc:In-Reply-To:References:Subject:List-ID:
	 List-Archive:List-Post:List-Unsubscribe:From;
	b=OKTtjuAXrmtPkmFruEfa4zjFugkaRkw8Y0EaM+NVgVIqskVq576XoYs1mAxiyOm1x
	 kGJmGPgCBuJ899apjXLq+i4QkPun904+fNsiKvZSbs9X+ec4Io+6h1hfDucZcggF0g
	 skyAYqzDjYilXvj7LSix0/D3ctiCMw2TrFwqGLBY=
